patches/blockprotocol+0.0.7.patch

```diff
@@ -33,49 +33,160 @@ index 20a9bbd..a849fa3 100644
+   index?: number | null;
+   path: string;
+-} & (SingleTargetLinkFields | AggregationTargetLinkFields);
++};
+ 
+ export type BlockProtocolLinkGroup = {
+   sourceAccountId?: string | null;
+@@ -279,14 +277,6 @@ export type BlockProtocolLinkGroup = {
+   links: BlockProtocolLink[];
+ };
+ 
+-export type BlockProtocolLinkedAggregation = {
+-  sourceAccountId?: string | null;
+-  sourceEntityId: string;
+-  sourceEntityVersionId?: string | null;
+-  sourceEntityTypeId?: string | null;
+-  path: string;
+-} & BlockProtocolAggregateEntitiesResult<BlockProtocolEntity>;
+-
+ export type BlockProtocolGetLinkAction = {
+   linkId: string;
+ };
+@@ -304,14 +294,7 @@ export type BlockProtocolCreateLinksFunction = {
    (actions: BlockProtocolCreateLinksAction[]): Promise<BlockProtocolLink[]>;
  };
  
-+export type BlockProtocolUpdateLinkedAggregationActionFragment = {
-+  // temporary identifiers for LinkedAggregations - to be replaced with a single id
-+  sourceAccountId?: string | null;
-+  sourceEntityId: string;
-+  path: string;
-+}
-+
- export type BlockProtocolUpdateLinksAction =
-   | { linkId: string; data: Pick<BlockProtocolLink, "index"> }
+-export type BlockProtocolUpdateLinksAction =
+-  | { linkId: string; data: Pick<BlockProtocolLink, "index"> }
 -  | ({
 -      // temporary identifiers for LinkedAggregations - to be replaced with a single id
 -      sourceAccountId?: string | null;
 -      sourceEntityId: string;
 -      path: string;
 -    } & { data: BlockProtocolAggregateOperationInput });
-+  | (BlockProtocolUpdateLinkedAggregationActionFragment & { data: BlockProtocolAggregateOperationInput });
++export type BlockProtocolUpdateLinksAction = { linkId: string; data: Pick<BlockProtocolLink, "index"> }
  
  export type BlockProtocolUpdateLinksFunction = {
    (actions: BlockProtocolUpdateLinksAction[]): Promise<BlockProtocolLink[]>;
-@@ -452,4 +459,4 @@ export type BlockProtocolProps = {
-   linkedAggregations?: BlockProtocolLinkedAggregation[];
+@@ -327,6 +310,73 @@ export type BlockProtocolDeleteLinksFunction = {
+   (actions: BlockProtocolDeleteLinksAction[]): Promise<boolean[]>;
+ };
+ 
++// ---------------------- LINKED AGGREGATIONS ------------------------- //
++
++export type SingleTargetLinkFields = {
++  destinationEntityId: string;
++  destinationEntityTypeId?: string | null;
++  destinationEntityVersionId?: string | null;
++};
++
++export type BlockProtocolLinkedAggregationInput = {
++  aggregationId: string;
++  sourceAccountId?: string | null;
++  sourceEntityId: string;
++  sourceEntityTypeId?: string | null;
++  sourceEntityVersionId?: string | null;
++  path: string;
++  operation: BlockProtocolAggregateOperationInput;
++};
++
++export type BlockProtocolLinkedAggregation = 
++  Omit<BlockProtocolLinkedAggregation, "operation">
++  & BlockProtocolAggregateEntitiesResult<BlockProtocolEntity>;
++
++export type BlockProtocolGetLinkedAggregationAction = {
++  sourceAccountId: string;
++  aggregationId: string;
++};
```

Comment:
the accountId is optional elsewhere (as not all apps may need it) – I suggest we could make `sourceEntityId` optional as well, in case the `aggregationId` is sufficient to identify the aggregation
```suggestion
+export type BlockProtocolGetLinkedAggregationAction = {
+  sourceAccountId?: string | null;
+  sourceEntityId?: string | null;
+  aggregationId: string;
+};
```

Reply:
Understood, addressed in https://github.com/hashintel/hash/pull/497/commits/800b6af44e74458c53b78ff9a8321b97e53dbc46